app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -404,4 +411,31 @@ object AppRepository {
     private fun resetChangesSeenFlag() =
             updateScheduleChangesSeen(false)
 
+    // region AlarmServices
```

Comment:
When I see section comments I always become a little uneasy. Often it signals that a class might have too many responsibilities. 
What is the advantage of this new delegate methods in AppRepository, instead of still directly inject AlarmService where its required?
You could however create a "inner class Alarm { .... }" to avoid section comments.

Reply:
Regions are picked up by the IDE to group certain functions in a class. IntelliJ/Android Studio shows the regions in the "Structure" panel.

I want to move the `AlarmServices` behind the `AppRepository` in order to have the latter as the central interface to manipulate "data" and to trigger UI updates. By moving `AlarmServices` here it becomes transparent to the caller that there is an actual `AlarmServices` class. The caller just talks to the `AppRepository`. This has the advantage that only one interface has to be injected into activities, fragments and such.

Can you please elaborate what you mean by `inner class Alarm {}`?

Reply:
Something like

```
inner class Alarm {
  fun scheduleEventAlarm(...) { ... }
}

// call site
appRepo.Alarm().scheduleEventAlarm(...)
```

Reply:
I would like to avoid exposing internals of a class (e.g. the `AppRepository` as much as possible. Therefore, I prefer not to let the caller dive into hierarchies which is considered [bad practice](https://hackernoon.com/object-oriented-tricks-2-law-of-demeter-4ecc9becad85).

Reply:
The cohesion of AppRepository will get very bad, if we continue to put everything related to data retrieval and manipulation in there. It's not bad to have segregated interfaces for different domains (either via distinct top level classes, or namespaces / nested classes).
According to domain driven design and classic layered software design, Service classes are not internal. They provide use case related operations and hide internal stuff (db access, transactions etc.). They sit between infrastructure layers (IO, UI). I could even imagine, that a Service would depend on any of the repository classes (as in classic DDD).

Reply:
Thank you for your thoughts. I decided not to unmangle this right now. See below. I changed the title and description of this pull request.